src/maxsubarray/MaxSubarray.js

```diff
@@ -13,4 +13,25 @@ export function getMaxSubarray(numbers, allowEmptySolution = true) {
     if (!allowEmptySolution && numbers.length === 0) {
         throw new Error("Input array must not be empty.")
     }
+    let currentSum = numbers[0];
+    let biggestSum = numbers[0];
+    let biggestArray = [ numbers[0] ];
+    let i = 0;
+    let j = 0;
+    for (let k = 1; k <= numbers.length - 1; k++) {
+        if (currentSum + numbers[k] < numbers[k]) {
+            currentSum = numbers[k];
+            i = k;
+            j = k;
+        } else {
+            currentSum = currentSum + numbers[k];
+            j++;
+        }
+        if (biggestSum < currentSum) {
+            biggestSum = currentSum;
+            biggestArray = numbers.slice(i, j + 1);
+        }
+    }
+
+    return allowEmptySolution ? (biggestSum > 0) ? biggestArray : [] : biggestArray;
```

Comment:
Nested ternary operators are hard to read and should be avoided. This can be done in a single line with just one ternary.